### DISCUSSION TITLE
[Question][Hw5] Folding/Iterating memory

### DISCUSSION THREAD

**Discussion by Standchen** [prosyslab-classroom/cs348-information-security#589] [Q&A]
Hello.
Implementing `filter` in `semantics.ml`, I guess that we need a way to iterate/fold the memory, since it's Map.
However, with given modules (`module Memory = Memory`, `module Value = Memory.Value`), we can't directly access the `module M = Map.Make (Variable)` in `module Memory` of `domain.ml`, so I'm not able to use any functions related to Map.

Is there any way to access this Map(`M`) somehow, or am I approaching in weird way?

Thank you in advance.

**Comment by KihongHeo** [comment]
You can define whatever you want in the Memory module and specify them in the MEMORY module type.

**Comment by Standchen** [comment]
Thank you!
